### PR TITLE
Allow partially applied constraint index to be recovered into tentative state on restart

### PR DESCRIPTION
**Pull request by MishaDemianenko**

### CODE REVIEW

File: community/kernel/src/test/java/org/neo4j/kernel/RecoveryIT.java [hunk 1]
```diff
@@ -226,7 +227,7 @@ public void shouldRecoverIdsCorrectlyWhenWeCreateAndDeleteANodeInTheSameRecovery
         recoveredDatabase.shutdown();
     }
 
-    @Test( timeout = 60_000 )
+    @Test//( timeout = 60_000 )
```
Comment (chrisvest) [inline community/kernel/src/test/java/org/neo4j/kernel/RecoveryIT.java:230]: I guess you forgot to undo this.

Comment (MishaDemianenko) [reply community/kernel/src/test/java/org/neo4j/kernel/RecoveryIT.java:230]: yeah yeah not done with all of the tests, for now I'm just utilizing CI resources :)

**Comment by tinwelint** [comment]
One concern I have is that TentativeConstraintIndexProxy is designed to "hide" failures and instead let the transaction finally enabling the constraint see them. It may be a theoretical problem, but there's a chance that failures (coming from duplicates) would eat more and more memory until a constraint rule gets created for it.

**Comment by MishaDemianenko** [comment]
Closing

**Pull request closed without merging**